parse/lex.go

```diff
@@ -508,7 +523,22 @@ func lexIdentifier(l *lexer) stateFn {
 			l.backup()
 
 			return l.emitItem(mk(typ, s))
-		case len(s) > 0 && isName(r):
+		case typ == 0:
```

Comment:
why not `itemError`?

Reply:
what if rename `itemError` to `itemNone`?

Reply:
we need itemError, it is used to handle errors.

Reply:
but i will change swithc to itemError with comment, that that is an zero value for item type

Reply:
`itemNone` would imply an error since items has to have some type 🤔
anyway treat this just as a suggestion for alternative